R/disease_progression.R

```diff
@@ -19,15 +19,27 @@ update_infection <- function(
 }
 
 create_progression_process <- function(
-  state,
-  from_state,
-  to_state,
-  rate,
-  infectivity,
-  new_infectivity
-  ) {
+    state,
+    from_state,
+    to_state,
+    rate,
+    infectivity,
+    new_infectivity
+) {
   function(timestep) {
-    to_move <- state$get_index_of(from_state)$sample(1/rate)
+    
+    # Retrieve the indices of all individuals in the to_move state:
+    index <- state$get_index_of(from_state)
+    
+    # If the length of rate is greater than 1 (when it's a variable):
+    if (length(rate) > 1) {
+      rate <- rate$get_values(index)
+    }
+    
+    # Sample the individuals to be moved into a new Bitset using the transition rate(s):
+    to_move <- index$sample(1/rate)
```

Comment:
I've been trying to trace back through the code to see if this is a rate or a probability and it caused me some confusion.

As used here the parameter `rate` is actually an average duration so is misnamed.
That duration gets converted to a rate in `1/rate` and provided as an argument to `$sample`. Would this be possible to clarify in the malariasimulation code @tbreweric?

@giovannic going deeper into this rabbit hole....I looked at the sample function in th individual package [here](https://github.com/mrc-ide/individual/blob/4bef32e1f2bb12796046467c88343902b5efc08c/inst/include/IterableBitset.h#L633-L653). That details an argument called `rate` which is actually a probability.

We should be careful here as in the code here `1/rate` is actually a rate and not a probability. The approximation only holds if the rate is small. For most cases this is probably fine, but I think at least the documentation and naming should be clear and precise. Sorry if this seems pedantic!

Reply:
Ah that's a good point on the individual side! I've mad an [issue](https://github.com/mrc-ide/individual/issues/196) to rename those variable names and docs to make it clearer. 

[Richard's utilities](https://github.com/mrc-ide/malariasimulation/blob/vivax_main/R/utils.R#L68-L74) (which assume an exponential process) may help make these conversions clearer on the malariasimulation side, if that's what you want to do.

Reply:
Not sure how best to proceed here as I think there are two options:
1. Change the name of the input in `create_progression_process()` to `average_residence_time `(or something better) so that you are then more obviously calculating the rate by taking the inverse of this, and then feed `$sample` a `rate`. The other instances where `create_progression_process()` is called do not use the arg name so wouldnt need to be amended I don't think. 

2. Change the inputs for `rate` in the `create_progression_process()` calls in `create_processes()` to `1/parameters$dx` so that we are feeding the function a rate.

With each of these, I'm wondering whether this PR is the place to make the changes or whether we make it the subject of it's own for clarity? What do you all think?

Reply:
@pwinskill @giovannic Is there a preference for what to do here? My instinct is to address this is a separate, small PR but happy to go with whichever you guys think is best!

Reply:
I have made an issue to address this (#290). I have also made a branch (bug/progression_process_rates) with a proposed fix (changing the rate inputs in the `create_progression_processes()` to `1/parameters$dX` and changing the `sample()` input to `rate `from `1/rate`.

I think if we merge this branch first (if everyone is satisfied with the changes), then I'll amend the branch Ive made to make it compatible with this one and then submit a PR for that. That should give us something clearer in the short term before the proposed changes to the `sample()` function in individual?

Reply:
Hi sorry for the delay on this.

Thank you for making a separate issue for this. I agree that it is out of scope for this PR, which should focus on SPC.

Let's carry on this conversation in #290